src/reducers/user_reducer.js

```diff
@@ -0,0 +1,34 @@
+import types from '../action/type'
+
+const DEFAULTSTATE={
+   user:null,
+   cart:null,
+   checkout:null,
+   loggedIn: false
+}
+export default (state = DEFAULTSTATE,action)=>{
+  switch(action.type){
+    case types.ACTIVE_USER_LOGIN:
+      return{
+        user:action.payload,
+        loggedIn:true
+      }
+    case types.USER_LOGIN:
+      return{
+        user:action.payload,
+        loggedIn: true
+      }
+    case types.LOG_OUT:
+      return{
+        DEFAULTSTATE
```

Comment:
You need the spread operator here, this will create a property named `DEFAULTSTATE` set to the `DEFAULTSTATE` objects contents.

It should be this:
```JAVASCRIPT
return {
    ...DEFAULTSTATE
}
```